redis/src/main/scala/zio/redis/RedisClusterExecutorLive.scala

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2021 John A. De Goes and the ZIO contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.redis
+
+import zio._
+import zio.redis.RedisClusterExecutorLive._
+import zio.redis.api.Cluster.AskingCommand
+import zio.redis.codec.StringUtf8Codec
+import zio.redis.options.Cluster._
+import zio.schema.codec.Codec
+
+import java.io.IOException
+
+private[redis] final case class RedisClusterExecutorLive(
+  clusterConnectionRef: Ref.Synchronized[ClusterConnection],
+  scope: Scope.Closeable
+) extends RedisExecutor {
+
+  def execute(command: Chunk[RespValue.BulkString]): IO[RedisError, RespValue] = {
+
+    def executeAsk(executorZIO: IO[RedisError.IOError, RedisExecutor], asking: Boolean = false) =
+      for {
+        executor <- executorZIO
+        _        <- ZIO.when(asking)(executor.execute(AskingCommand.resp((), StringUtf8Codec)))
+        res      <- executor.execute(command)
+      } yield res
+
+    def executeSafe(keySlot: Slot) = {
+      val recover = executeAsk(getExecutor(keySlot)).flatMap {
+        case e: RespValue.Error => ZIO.fail(e.toRedisError)
+        case success            => ZIO.succeed(success)
+      }.catchSome {
```

Comment:
No need for `flatMap` here, it can be `map`.

Reply:
Sry. I don't understand how it can be map, if i need zio to fail

Reply:
You are right, misread it.